Author: Nilesh Patra <npatra974@gmail.com>
Description: Use external libfmt by default
Last-Changed: Sun, May, 14 2020
Forwarded: not-needed
--- spdlog.orig/CMakeLists.txt
+++ spdlog/CMakeLists.txt
@@ -80,7 +80,7 @@
 
 # install options
 option(SPDLOG_INSTALL "Generate the install target" ${SPDLOG_MASTER_PROJECT})
-option(SPDLOG_FMT_EXTERNAL "Use external fmt library instead of bundled" OFF)
+option(SPDLOG_FMT_EXTERNAL "Use external fmt library instead of bundled" ON)
 option(SPDLOG_FMT_EXTERNAL_HO "Use external fmt header-only library instead of bundled" OFF)
 option(SPDLOG_NO_EXCEPTIONS "Compile with -fno-exceptions. Call abort() on any spdlog exceptions" OFF)
 
--- spdlog.orig/include/spdlog/tweakme.h
+++ spdlog/include/spdlog/tweakme.h
@@ -71,7 +71,7 @@
 // In this case spdlog will try to include <fmt/format.h> so set your -I flag
 // accordingly.
 //
-// #define SPDLOG_FMT_EXTERNAL
+#define SPDLOG_FMT_EXTERNAL 1
 ///////////////////////////////////////////////////////////////////////////////
 
 ///////////////////////////////////////////////////////////////////////////////
--- spdlog.orig/include/spdlog/fmt/bin_to_hex.h
+++ spdlog/include/spdlog/fmt/bin_to_hex.h
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <spdlog/tweakme.h>
 #include <cctype>
 
 //
--- spdlog.orig/include/spdlog/fmt/fmt.h
+++ spdlog/include/spdlog/fmt/fmt.h
@@ -4,7 +4,7 @@
 //
 
 #pragma once
-
+#include <spdlog/tweakme.h>
 //
 // Include a bundled header-only copy of fmtlib or an external one.
 // By default spdlog include its own copy.
@@ -24,4 +24,4 @@
 #else // SPDLOG_FMT_EXTERNAL is defined - use external fmtlib
 #include <fmt/core.h>
 #include <fmt/format.h>
-#endif
\ No newline at end of file
+#endif
--- spdlog.orig/include/spdlog/fmt/ostr.h
+++ spdlog/include/spdlog/fmt/ostr.h
@@ -7,7 +7,7 @@
 //
 // include bundled or external copy of fmtlib's ostream support
 //
-
+#include <spdlog/tweakme.h>
 #if !defined(SPDLOG_FMT_EXTERNAL)
 #ifdef SPDLOG_HEADER_ONLY
 #ifndef FMT_HEADER_ONLY
--- spdlog.orig/src/fmt.cpp
+++ spdlog/src/fmt.cpp
@@ -6,6 +6,7 @@
 #error Please define SPDLOG_COMPILED_LIB to compile this file.
 #endif
 
+#include <spdlog/tweakme.h>
 #if !defined(SPDLOG_FMT_EXTERNAL)
 #include <spdlog/fmt/bundled/format-inl.h>
 
